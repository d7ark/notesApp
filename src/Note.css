
.Note-foot {
    margin: 40px 0;
}

.Note-hr {
    margin: 0 20px;
}

.Note-row {
    margin-top: 20px;
}
