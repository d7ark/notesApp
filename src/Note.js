import { Link } from 'react-router-dom';
import React, { Component } from 'react';
import Remarkable from 'remarkable';

class Note extends Component {
  getRawMarkup() {
    const md = new Remarkable();
    return { __html: md.render(this.props.note.text) };
  }

  handleDelete = () => {
    const { note, onDelete } = this.props;
    onDelete({ variables: { id: note.id } });
  };

  render() {
    const { note } = this.props;
    const formatedDate = new Date(note.createdAt).toLocaleString();
    return (
      <div>
        <hr className="Note-hr" />
        <div className="Note-row row">
          <div className="nine columns">
            <div dangerouslySetInnerHTML={this.getRawMarkup()} />
          </div>
          <div className="three columns">
            <button onClick={this.handleDelete}>Delete</button>
            <Link to={`/notes/${note.id}/edit`}>
              <button>Edit</button>
            </Link>
          </div>
        </div>
        <Link to={`/notes/${note.id}`}>
          <p className="Note-foot">Created {formatedDate}</p>
        </Link>
      </div>
    );
  }
}

export default Note;
