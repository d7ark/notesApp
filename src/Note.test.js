import React from 'react';
import { render } from 'react-testing-library';
import { BrowserRouter as Router } from 'react-router-dom';

import Note from './Note';

const mockNote = {
  id: '123123',
  createdAt: new Date('08-08-1989'),
  text: 'Sample text without markdown.',
};

it('renders component without markdown correctly', () => {
  const { container } = render(<Router><Note note={mockNote}/></Router>);
  expect(container).toMatchSnapshot();
});

it('renders component with markdown correctly', () => {
  mockNote.text = 'Sample text **with** markdown';
  const { container } = render(<Router><Note note={mockNote}/></Router>);
  expect(container).toMatchSnapshot();
});

it('renders markdown correctly', () => {
  mockNote.text = 'Sample text **with** _many_ markdowns';
  const { container } = render(<Router><Note note={mockNote}/></Router>);
  const markdownDiv = container.querySelector('.dangerousInnerHTML');
  expect(markdownDiv).toMatchSnapshot();
});