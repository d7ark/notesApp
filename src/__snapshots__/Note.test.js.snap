// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component with markdown correctly 1`] = `
<div>
  <div>
    <hr
      class="Note-hr"
    />
    <div
      class="Note-row row"
    >
      <div
        class="nine columns"
      >
        <div
          class="dangerousInnerHTML"
        >
          <p>
            Sample text 
            <strong>
              with
            </strong>
            markdown
          </p>
        </div>
      </div>
      <div
        class="three columns"
      >
        <button>
          Delete
        </button>
        <a
          href="/notes/123123/edit"
        >
          <button>
            Edit
          </button>
        </a>
      </div>
    </div>
    <a
      href="/notes/123123"
    >
      <p
        class="Note-foot"
      >
        Created 
        8/8/1989, 12:00:00 AM
      </p>
    </a>
  </div>
</div>
`;

exports[`renders component without markdown correctly 1`] = `
<div>
  <div>
    <hr
      class="Note-hr"
    />
    <div
      class="Note-row row"
    >
      <div
        class="nine columns"
      >
        <div
          class="dangerousInnerHTML"
        >
          <p>
            Sample text without markdown.
          </p>
        </div>
      </div>
      <div
        class="three columns"
      >
        <button>
          Delete
        </button>
        <a
          href="/notes/123123/edit"
        >
          <button>
            Edit
          </button>
        </a>
      </div>
    </div>
    <a
      href="/notes/123123"
    >
      <p
        class="Note-foot"
      >
        Created 
        8/8/1989, 12:00:00 AM
      </p>
    </a>
  </div>
</div>
`;

exports[`renders markdown correctly 1`] = `
<div
  class="dangerousInnerHTML"
>
  <p>
    Sample text 
    <strong>
      with
    </strong>
    <em>
      many
    </em>
    markdowns
  </p>
</div>
`;
